components/header/navbar/searchpage/index.tsx

```diff
@@ -0,0 +1,292 @@
+import React, { useState, useEffect } from 'react'
+import Column from '@components/Column'
+import { SearchData, searchDrive, searchSiteContent } from '@lib/strapi/search'
+import { titleToAnchor } from '@utils/helpers'
+import ListCard from './ListCard'
+import SearchBar from './SearchBar'
+import { MdCancel } from 'react-icons/md'
+import { drive_v3 } from 'googleapis'
+import FileCard from './FileCard'
+
+type SearchOverlayProps = {
+  onClose: () => void
+  sessionToken?: string
+  setSideMenuOpen: (state: boolean) => void
+}
+
+const SearchOverlay = ({
+  onClose,
+  sessionToken,
+  setSideMenuOpen,
+}: SearchOverlayProps) => {
+  const [query, setQuery] = useState('')
+  const [fileSearch, setFileSearch] = useState(false)
+  const [searching, setSearching] = useState(false)
+  const [isFetched, setIsFetched] = useState(false)
+  const [fileResults, setFileResults] = useState<drive_v3.Schema$FileList>({
+    files: [],
+  })
+  const [results, setResults] = useState<SearchData>({
+    sectionData: [],
+    pageData: [],
+    privateSectionData: [],
+    privatePageData: [],
+  })
+
+  const clearResults = () => {
+    setResults({
+      sectionData: [],
+      pageData: [],
+      privateSectionData: [],
+      privatePageData: [],
+    })
+    setFileResults({ files: [] })
+    setIsFetched(false)
+  }
```

Comment:
As there is quite a lot of state in this component it might be worth it to introduce a [reducer](https://react.dev/learn/extracting-state-logic-into-a-reducer)